### ISSUE TITLE
Proposed solution for dealing with joins and nested beans

### ISSUE THREAD

**Issue by kshep92** [aaberg/sql2o#234]
Earlier today I needed to map the results of a query to a class that contained a sub-object. To my surprise

`query.addColumnMapping("rating_id", "rating.id")`

Did not work. So instead of going though the arduous task of using a `ResultSetHandler` and doing manual mapping, I created a class that creates an instance of a user-defined class based on the columns and values returned in a query. 

Take a look at the API below for the `RowSkimmer` class:

![skimmer](https://cloud.githubusercontent.com/assets/3780572/16352496/d6eb3702-3a3c-11e6-83bb-362dafc340aa.PNG)

As you can see the query returns fields for both a `Movie` class and a `Rating` class and the "skimmer" looks at all the returned results and parses out instances of these classes. 

There are some conventions though: 
- The target classes need to be Java Beans (having getters, setters and private fields) and 
- The column names returned need to match the fields declared in the target classes exactly (no underscores). 
- To extract nested beans, the columns should be returned in the format `[namespace]_[fieldName]`

It's still early, but so far this is working well for a project I'm currently working on. Do you guys think this could be useful to sql2o going forward?

Also, I am aware about the usage of `select r.id as \"rating.id\"` to achieve a similar result, but I just found it a bit ugly.

[Google Groups question](https://groups.google.com/forum/?no#!topic/sql2o/eucKt1vVh9o)

**Comment by aaberg** [comment]
I agree that the `RowSkimmer` seems to be a helpful and handy class to have available for situations as the one you are facing. But I don't think this api should belong as part of the sql2o api. The main reason for me saying this, is that you are effectively introducing a new mapper into the framework, which I don't think is a good idea.

**Comment by kshep92** [comment]
Fair enough. Has there been any any progress otherwise on getting joins to work with the current mapper?
